packages/zowe-explorer/src/commands/MvsCommandHandler.ts

```diff
@@ -71,49 +81,25 @@ export class MvsCommandHandler extends ZoweCommandProvider {
             }
         }
         if (!session) {
-            const allProfiles = profiles.allProfiles;
-            const profileNamesList = ProfileManagement.getRegisteredProfileNameList(Definitions.Trees.MVS);
-            if (profileNamesList.length) {
-                const quickPickOptions: vscode.QuickPickOptions = {
-                    placeHolder: vscode.l10n.t("Select the Profile to use to submit the command"),
-                    ignoreFocusOut: true,
-                    canPickMany: false,
-                };
-                const sesName = await Gui.showQuickPick(profileNamesList, quickPickOptions);
-                if (sesName === undefined) {
-                    Gui.showMessage(vscode.l10n.t("Operation Cancelled"));
-                    return;
-                }
-                profile = allProfiles.filter((temprofile) => temprofile.name === sesName)[0];
-                if (!node) {
-                    await profiles.checkCurrentProfile(profile);
-                }
-                if (profiles.validProfile !== Validation.ValidationType.INVALID) {
-                    session = ZoweExplorerApiRegister.getMvsApi(profile).getSession();
-                } else {
-                    Gui.errorMessage(vscode.l10n.t("Profile is invalid"));
-                    return;
-                }
-            } else {
-                Gui.showMessage(vscode.l10n.t("No profiles available"));
+            profile = await this.selectNodeProfile(Definitions.Trees.MVS);
+            if (!profile) {
                 return;
             }
         } else {
             profile = node.getProfile();
         }
         try {
-            if (profiles.validProfile !== Validation.ValidationType.INVALID) {
+            if (this.profileInstance.validProfile !== Validation.ValidationType.INVALID) {
                 const commandApi = ZoweExplorerApiRegister.getInstance().getCommandApi(profile);
                 if (commandApi) {
-                    let command1: string = command;
                     if (!command) {
-                        command1 = await this.getQuickPick(session && session.ISession ? session.ISession.hostname : "unknown");
+                        command = await this.getQuickPick([session && session.ISession ? session.ISession.hostname : "unknown"]);
                     }
-                    await this.issueCommand(profile, command1);
-                } else {
-                    Gui.errorMessage(vscode.l10n.t("Profile is invalid"));
-                    return;
+                    await this.issueCommand(profile, command);
                 }
+            } else {
+                Gui.errorMessage(vscode.l10n.t("Profile is invalid"));
+                return;
```

Comment:
If the profile is invalid, should we provide an option in the error message to pick another profile to run the command with instead? It may be perceived as inconvenient if the user has to type the command again after selecting a profile that is not valid.

Reply:
Since this "profile is invalid" message happens early in the process, I don't believe it's necessary to reprompt for a profile since they haven't types any commands any more information.

Unix is the exception though.
I've handled it in 7833f740a9bee6d27c5380f50446502300b0064b

But it only validates the nodeProfile, i.e. ZOS connection profile (Zosmf, FTP, RSE, ...).
We don't validate the SSH profile but maybe we should 😉 
See 88e064b0abcbd461bbdc795a5eb85fb075148ad4

Reply:
Moving the SSH validation to the SDKs to avoid having to import ssh2 directly here 🙏